helm/temporal-worker-controller/templates/deployment.yaml

```
@@ -0,0 +1,85 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: temporal-worker-controller
+  namespace: {{ .Values.namespace }}
+  labels:
+    control-plane: controller-manager
+    app.kubernetes.io/name: deployment
+    app.kubernetes.io/instance: temporal-worker-controller
+    app.kubernetes.io/component: manager
+    app.kubernetes.io/created-by: worker-controller
+    app.kubernetes.io/part-of: worker-controller
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
```

I think there are a few adjustments to make based on recommendations from https://helm.sh/docs/chart_best_practices/labels/#standard-labels

`app.kubernetes.io/name` should be the chart name
`app.kubernetes.io/instance` should be `$.Release.Name` (needs to be unique to every instance created within the k8s namespace)
`app.kubernetes.io/component` lgtm
`app.kubernetes.io/created-by` delete (it's [deprecated](app.kubernetes.io/created-by))
`app.kubernetes.io/part-of` delete (it's optional, and the typical use case is to indicate that something like a generic cache or database is part of a higher level application like "wordpress")
`app.kubernetes.io/managed-by` lgtm

---

Actually every resource should just get its labels from the helper for consistency.

```suggestion
{{ include "temporal-worker-controller.labels" $ | indent 4 }}
```

---

Thanks so much for typing this out! I just implemented that guidance / corrected the silly AI labels before I saw your comments. The best practice links you're posting are v helpful too